### PR TITLE
Tech Tree

### LINKED ISSUE #3: Suggestion: Tech Tree
It would be very helpful to see all steps necessary to craft something on a single page. You could show a tech tree [similar to this](https://onehouronelife.com/forums/viewtopic.php?pid=2201#p2201). Perhaps add a "Tech Tree" button underneath the object info.

Some objects have multiple "from" transitions. For these you could create a formula for selecting the most common case. For example, to make a [Long Straight Shaft](https://kazetsukai.github.io/onetech/#67/Long-Straight-Shaft) you want to show crafting a sharp stone with a straight branch.
* Ignore timed transitions
* Ignore empty hand transitions
* Avoid circular dependencies
* Prefer transitions that result in only that object

Some special coding might still be required for certain cases. I can work on a pull request for this if you like.

### PR DESCRIPTION
**Pull request by ryanb**
**Note:** This extends the [Separate JSON Files PR](https://github.com/Kazetsukai/onetech/pull/10), so please accept that one first.

* Updates the process script to add a `techTree` attribute to the object data which is 3 levels deep.
* Adds Tech Tree button to the object inspector
* Tech Tree view which displays the objects in a tree view
* Arrows allow one to expand the tree

screen shot 2018-03-17 at 5 39 14 pm

screen shot 2018-03-17 at 4 50 27 pm

Closes #3

Feel free to make suggestions or change something.

### CODE REVIEW

File: process/src/SpriteProcessor.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+      width,
+      height,
+      (newCanvas.width - width) / 2,
+      (newCanvas.height - height) / 2,
+      width,
+      height
+    );
+
+    // Debug
+    this.context.beginPath(); //
```
Comment (Kazetsukai) [inline process/src/SpriteProcessor.js]: Should this still be in here?

Comment (ryanb) [reply process/src/SpriteProcessor.js]: I left it in just in case. Feel free to remove.

**Comment by Kazetsukai** [comment]
Looks really good. Merging now, if I find anything wrong I will make new commit to change.

**Comment by Kazetsukai** [comment]
Yep working nicely. Should be live as soon as GitHub pages catches up (I might slap this all behind cloudflare or maybe move to AWS if it gets more use)

Did a tiny tweak to the CSS, using calc to get the vertical lines to line up

**Pull request merged**